### PR TITLE
Feature/edit

### PR DESCRIPTION
**Pull request by ypayel**

### CODE REVIEW

File: src/pages/EditBooks/EditBooks.tsx [hunk 1]
```diff
@@ -1,109 +1,149 @@
...
-                    
-                </div>
-            ) : (
-                error ? <p>{error}</p> : <p>Loading...</p>
-            )}
+            <form onSubmit={handleSubmit} className="form">
+                {book ? (
+                    <div className="inputs">
+                        <label htmlFor="input-title">
+                            <p>
```
Comment (szu-szy) [inline src/pages/EditBooks/EditBooks.tsx:67]: <p> there is unnecessary

File: src/pages/EditBooks/EditBooks.tsx [hunk 2]
```diff
@@ -1,109 +1,149 @@
...
+                                <input
+                                    className="input-title"
+                                    value={book.title}
+                                    name="title"
+                                    onChange={handleEdit}
+                                />
+                            </p>
+                        </label>
+                        <label htmlFor="input-authors">
+                            <p>
```
Comment (szu-szy) [inline src/pages/EditBooks/EditBooks.tsx:78]: as above

File: src/pages/EditBooks/EditBooks.tsx [hunk 3]
```diff
@@ -1,109 +1,149 @@
...
+                                    className="input-authors"
+                                    type="text"
+                                    value={book.authors ? book.authors.join(", ") : ""}
+                                    name="authors"
+                                    onChange={handleEdit}
+                                />
+                            </p>
+                        </label>
+                        <label htmlFor="input-pageCount">
+                            <p>
```
Comment (szu-szy) [inline src/pages/EditBooks/EditBooks.tsx:90]: as above

File: src/pages/EditBooks/EditBooks.tsx [hunk 4]
```diff
@@ -1,109 +1,149 @@
...
+                                    className="input-pageCount"
+                                    type="text"
+                                    value={book.pageCount}
+                                    name="pageCount"
+                                    onChange={handleEdit}
+                                />
+                            </p>
+                        </label>
+                        <label htmlFor="input-isbn">
+                            <p>
```
Comment (szu-szy) [inline src/pages/EditBooks/EditBooks.tsx:102]: as above

File: src/pages/EditBooks/EditBooks.tsx [hunk 5]
```diff
@@ -1,109 +1,149 @@
...
+                                    className="input-isbn"
+                                    type="text"
+                                    value={book.isbn}
+                                    name="isbn"
+                                    onChange={handleEdit}
+                                />
+                            </p>
+                        </label>
+                        <label htmlFor="input-id">
+                            <p>
```
Comment (szu-szy) [inline src/pages/EditBooks/EditBooks.tsx:114]: as above

File: src/pages/EditBooks/EditBooks.tsx [hunk 6]
```diff
@@ -1,109 +1,149 @@
...
+                }
+                const updatedBook = { ...prev, [name]: value };
+                return updatedBook;
+            }
+            return prev;
         });
-      };
+    };
 
     const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
```
Comment (szu-szy) [inline src/pages/EditBooks/EditBooks.tsx:56]: right now book after submit dont request updated data to server, is it okay? (just asking)

**Review by szu-szy — CHANGES_REQUESTED** [review]
please check comments

File: src/pages/EditBooks/EditBooks.tsx (hunk 6 above)
Comment (ypayel) [reply src/pages/EditBooks/EditBooks.tsx:56]: yes

**Review by szu-szy — APPROVED** [review]

**Pull request merged**